### ISSUE TITLE
GDScript String.contains() method returns false for an empty string

### ISSUE THREAD

**Issue by lamdevhs** [godotengine/godot#82473]
### Godot version

v4.1.1.stable.arch_linux

### System information

Godot v4.1.1.stable unknown - Arch Linux #1 SMP PREEMPT_DYNAMIC Wed, 13 Sep 2023 09:32:00 +0000 - Vulkan (Compatibility) - Mesa Intel(R) HD Graphics 4600 (HSW GT2) () - Intel(R) Core(TM) i5-4590 CPU @ 3.30GHz (4 Threads)

### Issue description

 In GDScript, calling foo.contains("") for any string 'foo' (including the empty string) returns false. Philosophy aside I think that's not a desirable default behavior at all. Most if not all programming languages assert the empty string to be contained in any string (especially the empty string itself!). Otherwise we end up in a weird NaN != NaN situation.
In Python:
print("" in "" and "" in "foo") -> prints out 'True'
In JavaScript:
console.log("".includes("") && "foo".includes("")) -> prints out 'true'

### Steps to reproduce

print("foo".contains("")) # should print true, but prints false
print("".contains("")) # should print true, but prints false

### Minimal reproduction project

N/A

**Comment by lamdevhs** [comment]
Found the problem: `foo.contains(bar)` is just an alias for `foo.find(bar) != -1`, and the implementation of `find()` starts with this code:
https://github.com/godotengine/godot/blob/4c3dc26367518e006f8555c12f5d2df0b8a28192/core/string/ustring.cpp#L2982-L2988
Either that `if` statement is removed, as it's just optimization, or it should be replaced with:
```c++
if (src_len == 0) {
    return 0; // empty string found immediately
}
else if (len == 0) {
    return -1; // cannot find a non-empty string in an empty string
}
```

**Comment by SaNeOr** [comment]
I agree, and I've tested it locally.

**Comment by motadv** [comment]
Completely agree! This makes no sense at all, even if you think mathematically! In Set Theory, every set contains the empty set.

**Comment by lamdevhs** [comment]
Yep, sadly it was never fixed... I dunno if anyone's ever gonna bother to come back to this issue (and the issues around it in the code) and fix it properly (you can check out the discussion in the pull request above if you wanna know why it was never fixed).

Personally I just made my own wrapper function to use instead of the broken built-in one.
